server/controllers/medical/patients/index.js

```diff
@@ -246,6 +247,59 @@ function lookupPatient(patientUuid) {
     });
 }
 
+
+/**
+ * @method updatePatientDebCred
+ *
+ * @description
+ * This function is used to update the text value of the creditor 
+ * and debitor tables in case the patient's name was changed
+ *
+ * @param {String} patientUuid - the patient's unique id hex string
+ */
+
+function updatePatientDebCred(patientUuid) {
+  // convert uuid to database usable binary uuid
+  const buid = db.bid(patientUuid);
+
+  const sql = `
+    SELECT BUID(debtor.uuid) AS debtorUuid, BUID(employee.creditor_uuid) AS creditorUuid, patient.display_name
+    FROM debtor
+    JOIN patient ON patient.debtor_uuid = debtor.uuid
+    LEFT JOIN employee ON employee.patient_uuid = patient.uuid
+    WHERE patient.uuid = ?
+  `;
+
+  return db.exec(sql, buid)
+    .then((row) => {      
+      const debtorUuid = db.bid(row[0].debtorUuid);
+      const creditorUuid =  db.bid(row[0].creditorUuid);
+
+      const debtorText = {
+        text : `Debiteur [${row[0].display_name}]`
+      };
+
+      const creditorText = {
+        text : `Crediteur [${row[0].display_name}]`,
+      };
+
+      const updateCreditor = `UPDATE creditor SET ? WHERE creditor.uuid = ?`;
+      const updateDebtor = `UPDATE debtor SET ? WHERE debtor.uuid = ?`;
+
+      const transaction = db.transaction();
+
+      transaction
+        .addQuery(updateDebtor, [debtorText, debtorUuid])
+        .addQuery(updateCreditor, [creditorText, creditorUuid]);
+
+      transaction.execute()
+      .then((res) => {
```

Comment:
No need for this `.then` statement.  You should probably just do `return transaction.execute()`.